cpp/dolfinx/fem/assemble_vector_impl.h

```diff
@@ -54,7 +54,8 @@ void _lift_bc_cells(
     int cstride, const std::span<const std::uint32_t>& cell_info,
     const std::span<const T>& bc_values1,
     const std::span<const std::int8_t>& bc_markers1,
-    const std::span<const T>& x0, double scale)
+    const std::span<const std::int8_t>& bc_cells1, const std::span<const T>& x0,
```

Comment:
Pass spans by copy.

This function signature is a bit muddled now. We pass `cells` and `bc_cells1`. It would be neater to pass just the cells to be (1) assembled over and (2) which also have bcs applied.

Reply:
> Pass spans by copy.

Should I change this for `bc_values1`, `bc_markers1` as well?